### ISSUE TITLE
Add penalty on first derivative to loss function

### ISSUE THREAD

**Issue by alizma** [METHODS-Group/DRDMannTurb#7]
In addition to the regularization term and the MSE term in the loss function, there is currently an additional term penalization on the second derivative of the learned spectra. This is given in the form (equation 27) 
$$\text{Pen} [\theta] \propto \sum ^4 _{i = 1} \left| \left| \text{ReLU} \left( \frac{\partial ^2 \log |J_i (\cdot, \theta) |}{(\partial \log k_1 )^2} \right) \right| \right| ^2  $$
(this exists to encourage convex curves and minimize oscillations by penalizing curvature of the output space)

Now in addition to this, we want another penalty term on the first derivative. This will be in the form 
$$\text{Pen} [\theta] \propto \sum ^4 _{i = 1} \left| \left| \text{GELU} \left( \frac{\partial \log |J_i (\cdot, \theta) |}{\partial \log k_1 } \right) \right| \right| ^2  $$
and the aim is to penalize the spurious stretching of the output spectra, which has been observed in models that train past the first paper's loss of 1e-3. This stretching is most evident in the sharp derivative of the output spectra. 

The proposed penalty term introduces an additional hyperparameter into the loss function and this should be chosen with care to ensure a convex combination of terms in the loss function. 

Since higher-order derivatives are computed for the original second-order penalty, the intermediate derivatives can be used to compute the second penalty term.
